frontend/javascript/page_navigation.js

```diff
@@ -16,6 +16,73 @@ export function enableScrollToTop() {
   });
 }
 
+export function setupSidebarItemEventListeners() {
+  const sidebarItems = document.querySelectorAll('.sidebar-item-h2');
+
+  sidebarItems.forEach(function(item) {
+    item.addEventListener('mouseenter', function() {
+      item.style.color = '#34d399';
+    });
+    item.addEventListener('mouseleave', function() {
+      item.style.color = 'gray';
+    });
+    item.addEventListener('click', function() {
+      const currentVersion = item.textContent;
+      const headings = document.querySelectorAll('h2');
+      let targetElement = null;
+      headings.forEach(function(heading) {
+        if (heading.textContent.trim() === currentVersion.trim()) {
+          targetElement = heading;
+        }
+      });
+      if (targetElement) {
+        window.scrollTo(0, targetElement.offsetTop - 100);
+      }
+    });
+  });
+}
+
+export function setupSidebar() {
+  const h1Elements = document.querySelectorAll('h1');
+  const h2Elements = document.querySelectorAll('h2');
+  const sidebar = document.querySelector('.sidebar');
+
+  sidebar.style.position = 'fixed';
+  sidebar.style.top = '100px';
+  sidebar.style.right = '50px';
+  sidebar.style.width = 'fit-content';
+  sidebar.style.maxWidth = '300px';
+  sidebar.style.height = '100%';
+
+  function appendSidebarItem(textContent, tagName) {
+    const newItem = document.createElement('p');
```

Comment:
these should be anchor tags so that they are accessible. That would have the added advantage that you then don't need to apply custom colors on mouseenter and mouseleave.